app/SettingsScreen.js

```diff
@@ -0,0 +1,127 @@
+import React, {Component} from 'react';
+import { View, Text, Button, Switch, StyleSheet, Alert, TextInput, AsyncStorage } from 'react-native';
+
+export default class SettingsScreen extends Component{
+	state = {
+		blackBackground: false,
+		username: '',
+	}
+
+	componentDidMount(){
+		this.storageToState();
+	}
+
+	storageToState = async () => { 
+		try {
+  			const value = await AsyncStorage.getItem('@DemoRNStore:settings_background');
+  			if (value !== null){
+    			this.setState({blackBackground: value == 'true'});
+  			}
+		} catch (error) {
+			console.error(error);
+  			// Alert.alert(JSON.stringify(error));
+		}
+		try {
+  			const value = await AsyncStorage.getItem('@DemoRNStore:settings_username');
+  			if (value !== null){
+    			this.setState({username: value});
+  			}
+		} catch (error) {
+			console.error(error);
+  			// Alert.alert('get username error');
+		}
+	}
+
+	toggleSwitch = (value) => {
+		this.setState({blackBackground: value});
+		this.storeBackground(value);
+	}
+
+	storeBackground = async (value) => {
+		try {
+  			await AsyncStorage.setItem('@DemoRNStore:settings_background', value.toString());
+		} catch (error) {
+			console.error(error);
+  			// Alert.alert('put background error');
+		}
+	}
+
+	onUserNameChanged = (text) => {
+		this.setState({username: text});
+		this.storeUserName(text);
```

Comment:
setState function has it's callback, so if you need something to happen exactly after state change, better option is to write: this.setState({ username: text }, () => this.storeUserName(text));